docs/modules/airflow/pages/usage-guide/applying-custom-resources.adoc

```diff
@@ -0,0 +1,73 @@
+= Applying Custom Resources
+
+Airflow can be used to apply custom resources from within a cluster. An example of this could be a SparkApplication job that is to be triggered by Airflow. The steps below describe how this can be done.
+
+== Define an in-cluster Kubernetes connection
+
+An in-cluster connection can either be created from within the Webserver UI (note that the "in cluster configuration" box is ticked):
+
+image::airflow_connection_ui.png[Airflow Connections]
+
+Alternatively, the connection can be https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html[defined] by an environment variable in URI format:
+
+[source]
+AIRFLOW_CONN_KUBERNETES_IN_CLUSTER: "kubernetes://?__extra__=%7B%22extra__kubernetes__in_cluster%22%3A+true%2C+%22extra__kubernetes__kube_config%22%3A+%22%22%2C+%22extra__kubernetes__kube_config_path%22%3A+%22%22%2C+%22extra__kubernetes__namespace%22%3A+%22%22%7D"
+
+This can be supplied directly in the custom resource for all roles (Airflow expects configuration to be common across components):
+
+[source,yaml]
+----
+include::example$example-airflow-incluster.yaml[]
+----
+
+== Define a cluster role for Airflow to create SparkApplication resources
+
+Airflow cannot create or access SparkApplication resources by default - a cluster role is required for this:
+
+[source,yaml]
+----
+include::example$example-airflow-spark-clusterrole.yaml[]
+----
+
+and a corresponding cluster role binding:
+
+[source,yaml]
+----
+include::example$example-airflow-spark-clusterrolebinding.yaml[]
+----
+
+== DAG code
+
+Now for the DAG itself. The job to be started is a simple Spark job that calculates the value of pi:
+
+[source,yaml]
+----
+include::example$example-pyspark-pi.yaml[]
+----
+
+This will called from within a DAG by using the connection that was defined earlier. It will be wrapped by the `KubernetesHook` that the Airflow Kubernetes provider makes available. There are two classes that are used to:
```

Comment:
```suggestion
This will called from within a DAG by using the connection that was defined earlier. It will be wrapped by the `KubernetesHook` that the Airflow Kubernetes provider makes available [here.](https://github.com/apache/airflow/blob/main/airflow/providers/cncf/kubernetes/operators/spark_kubernetes.py) There are two classes that are used to:
```